client.py

```diff
@@ -405,32 +432,36 @@ def filter_hg_commit_data(repository_name, repository_url):
     :param repository_url: Link of the repository, eg: https://hg.mozilla.org/build/nagios-tools/
     :return: Returns a dictionary that contains the commits in the provided hg_repository_name
     """
-    request_url = repository_url + "json-log"
+    link = generate_hg_pushes_link(repository_name)
+    data = json.loads(requests.get(link).text)
+    last_push_id = data.get("lastpushid")
     hg_repo_data = {}
     commit_number = 0
     print("\nWorking on repo:", repository_name)
-    hg_repo_data.update({commit_number: {"lastChecked": str(datetime.utcnow())}})
-    data = json.loads(requests.get(request_url).text)
-    commit_number += 1
-
-    for entry in data["changesets"]:
-        node = entry["node"]
-        url = repository_url + "pushloghtml?changeset=" + node[:12]
-        commiter = entry["user"]
-        commiter_name = commiter.split('<')[0]
-        commiter_email = extract_email(commiter)
-        commit_message = entry["desc"]
-        message = re.sub("[*\n\r]", " ", commit_message)
-        date = entry["date"]
-        hg_repo_data.update({commit_number: {
-            "sha": node,
-            "url": url,
-            "commiter_name": commiter_name,
-            "commit_email": commiter_email,
-            "commit_message": message,
-            "commit_date": datetime.utcfromtimestamp(date[0]).strftime('%Y-%m-%d %H:%M:%S')
-        }})
-        commit_number += 1
+    hg_repo_data.update({commit_number: {"last_push_id": last_push_id}})
+    hg_repo_data2 = {}
+    data = json.loads(requests.get(generate_hg_pushes_link(repository_name)).text)
+    for key in data.get("pushes"):
+        changeset_number = key
+        changeset_pusher = data.get("pushes").get(key).get("user")
+        date_of_push = data.get("pushes").get(key).get("date")
+        for keys in data.get("pushes").get(key).get("changesets"):
+            author = keys.get("author")
+            desc = keys.get("desc")
+            files_changed = []
+            commit_number += 1
+
+            for entry in keys.get("files"):
+                files_changed.append(entry)
+            hg_repo_data2.update({commit_number: {"commiter_name": author,
+                                                  "commit_message": desc,
+                                                  "files_changed": files_changed}})
+
+            hg_repo_data.update({changeset_number: {
+                "Pusher": changeset_pusher,
+                "Date Of Push": time.strftime('%Y-%m-%d %H:%M:%S', time.localtime(date_of_push)),
+            }})
+            hg_repo_data.update(hg_repo_data2)
```

Comment:
You can use 'json_writer(repository_name, new_commits)' function to append the new commits to the json file.
param:
  - repository_name -> repository_name
  - new_commits -> hg_repo_data